### DISCUSSION TITLE
Question on Phidget VINT Hub0000_0 for PWM output command fan controller

### DISCUSSION THREAD

**Discussion by coffeeguru** [artisan-roaster-scope/artisan#1064] [Q&A] [answered]
Hi, I'm looking for help!  Been trying to get the [Phidget VINT HUB](https://www.phidgets.com/?tier=3&catid=2&pcid=1&prodid=643) to control fan speed with a [dimmer module](https://www.amazon.com/110V-240V-Controller-Regulating-Regulator-Compatible/dp/B0BC297G4B/ref=sr_1_2?crid=2UEYB75XYQUB0&keywords=pwm+dimmer+ac&qid=1672047863&sprefix=pwm+dimmer+ac%2Caps%2C166&sr=8-2) via PWM.
I can successfully use my PC to control the fan, but only with Phidget Control Panel and not with Artisan 2.8.2.
Through the Phidget Control Panel I can verify which port hub I'm on and successfully use the duty cycle to turn on and control the dimmer module, so it seems properly wired.  
![Screenshot 2022-12-26 002311](https://user-images.githubusercontent.com/1817698/209529425-21efb2b4-5459-446d-be0c-f1cdbfae2aae.png)

However on the same PC with nothing else changed, Artisan I doesn't trigger the PWM dimmer module.

I carefully read the Phidgets Various Modules and More Phidgets documentations, also saw a similar [github discussion](https://github.com/artisan-roaster-scope/artisan/discussions/804) and I've tried many of those solutions but still can't get Artisan to trigger a Hub0000_0 output.

I've tired different ports on the hub and addressed them accordingly, deactivated the Phidget Control Panel, updated the Phidget Hub firmware, reinstalled phidget drivers, updated to Artisan 2.8.2 and also tried various slider and button commands to produce a PWM out of the Hub. 

![Screenshot 2022-12-26 005625](https://user-images.githubusercontent.com/1817698/209530112-7ecfbf4e-4579-417c-b18e-94d15e490d5b.png)

![Screenshot 2022-12-26 004512](https://user-images.githubusercontent.com/1817698/209527450-a2b72ab8-5541-4e74-b9a0-33053e706a69.png)
Additionally, Artisan is able to read the [Phidget Air Sensor](https://www.phidgets.com/?prodid=98) attached to one of the [HUB0000_0](https://www.phidgets.com/?prodid=643) ports so there is communication between Artisan and the Hub0000_0.  

I'm scratching my head but soon be pulling out my hair.  I feel like it's down to my commands mistyped or something silly like a box I didn't tick.
Appreciate any insight and suggestions!  Thanks.
Raymond

**Comment by MAKOMO** [comment]
Your `outhub` command looks ok. You can use `outhub(1,{})` with the placeholder `{}`replaced by the value of the button (or slider). I just checked and this works for me thus on a HUB0001 (I don't have access to a HUB0000 anymore). Did you close the Phidget Control Panel before starting Artisan and connecting it to your hardware? Only one app (the Control Panel counts as one) can access the hardware directly at the same time (if you do not use the Phidget Network Service and connect via the (local) network). Could you turn debug mode ON (hold CTRL/ALT keys while clicking the plus icon), test again and then just hold the ALT key and click again. An Email should open with the file `artisan_device.log` attached. Rename this file into artisan_device.txt and attach it here on Github by drag-and-drop to a message box.

**Comment by coffeeguru** [comment, marked as answer]
It's working now!
As you recommended, I created a debug log and it has a lot of errors.  
[artisan_device.txt](https://github.com/artisan-roaster-scope/artisan/files/10304660/artisan_device.txt)

While adding back devices one at a time the error came back on the [Phidget Air Sensor 1136](https://www.phidgets.com/?tier=3&catid=7&pcid=5&prodid=109). 
I which is attached to the Hub0000_0.
![b1](https://user-images.githubusercontent.com/1817698/209579916-d9f6a8ad-5527-481f-94f0-e6a33c2a6abd.png)
Changing device assignment from 'Phidget HUB IO 1' to 'Phidget HUB IO 0' fixed the issue!
Grrr, I had it on Phidget HUB IO 1 all this time, but the column was truncated so I didn't notice, AND the Phidget 1136 was working fine with that setting so I didn't think it was an assignment issue.

Thank you so much, you saved me!  You are like my Santa Claus!  Truly appreciate your help.
Raymond

**Reply by MAKOMO** [reply]
Oh right. Forgot on this. Those Phidget device channels attach ports in pairs and those attached ports are then blocked for any other use and thus cannot be attached to send out a PWM signal. Good you resolved this.
